src/Npgsql/NpgsqlCommand.cs

```diff
@@ -1307,15 +1307,15 @@ internal void FixupRowDescription(RowDescriptionMessage rowDescription, bool isF
                 rowDescription[i].FormatCode = (UnknownResultTypeList == null || !isFirst ? AllResultTypesAreUnknown : UnknownResultTypeList[i]) ? FormatCode.Text : FormatCode.Binary;
         }
 
-        void LogCommand(int connectorId)
+        void LogCommand(int connectorId, bool excludeParametersFromLogs)
         {
             var sb = new StringBuilder();
             sb.AppendLine("Executing statement(s):");
             foreach (var s in _statements)
             {
                 sb.Append("\t").AppendLine(s.SQL);
                 var p = s.InputParameters;
-                if (NpgsqlLogManager.IsParameterLoggingEnabled && p.Count > 0)
+                if (NpgsqlLogManager.IsParameterLoggingEnabled && p.Count > 0 && excludeParametersFromLogs == false)
```

Comment:
This means that NpgsqlLogManager.IsParameterLoggingEnabled has to be enabled regardless of what's on the connection string - that seems to make the new connection string option pretty useless. I think it should be possible to simply enable parameter logging via one or the other mechanism.